ui/app/AppLayouts/Onboarding2/OnboardingFlow.qml

```diff
@@ -243,13 +251,35 @@ SQUtils.QObject {
         onReloadKeycardRequested: root.reloadKeycardRequested()
         onCreateProfileWithEmptyKeycardRequested: keycardCreateProfileFlow.init()
         onKeycardFactoryResetRequested: root.keycardFactoryResetRequested()
+        onUnlockWithPukRequested: {
+            const flow = root.getUnlockWithPukFlow()
+            flow.finished.connect(() => {
+                d.flow = Onboarding.SecondaryFlow.LoginWithKeycard
+                d.pushOrSkipBiometricsPage()
+            })
+        }
 
         onFinished: {
             d.flow = Onboarding.SecondaryFlow.LoginWithKeycard
             d.pushOrSkipBiometricsPage()
         }
     }
 
+    UnlockWithPukFlow {
+        id: unlockWithPukFlow
+
+        stackView: root.stackView
+        keycardState: root.keycardState
+        tryToSetPukFunction: root.tryToSetPukFunction
+        remainingAttempts: root.remainingPukAttempts
+
+        keycardPinInfoPageDelay: root.keycardPinInfoPageDelay
+
+        onReloadKeycardRequested: root.reloadKeycardRequested()
+        onKeycardPinCreated: (pin) => root.keycardPinCreated(pin)
+        onKeycardFactoryResetRequested: root.keycardFactoryResetRequested()
+    }
```

Comment:
I think that structurally it would be cleaner to keep the `UnlockWithPukFlow` one level above. Than `getUnlockWithPukFlow` (looking like workaround) is not needed. We also don't need to pass additional puk-related params there, what is needed is exposing `unlockWithPukRequested` to initialize the puk flow higher in the hierarchy.

Reply:
Yeah... open to ideas :) This was the only to be able to reuse the flow from 2 different places with 2 different ends (the `finished()` signal needs to be handled differently)

Reply:
Yeah, looks good, something along these lines!